module/zfs/zfs_ioctl.c

```diff
@@ -4112,14 +4112,16 @@ zfs_check_settable(const char *dsname, nvpair_t *pair, cred_t *cr)
 	{
 		spa_feature_t feature;
 		spa_t *spa;
-		uint64_t intval;
 		int err;
 
 		/* dedup feature version checks */
 		if (prop == ZFS_PROP_DEDUP &&
 		    zfs_earlier_version(dsname, SPA_VERSION_DEDUP))
 			return (SET_ERROR(ENOTSUP));
 
+		if (nvpair_type(pair) == DATA_TYPE_BOOLEAN)
+			break;
```

Comment:
nit: could we instead structure this in the same way as the other case blocks.  The explicit check for `DATA_TYPE_BOOLEAN` I find a little non-intuitive.

```c
                if (nvpair_type(pair) == DATA_TYPE_UINT64 &&
                    nvpair_value_uint64(pair, &intval) == 0) {
                        ...
                }
```